### PR TITLE
Refactor start broadcasting and update precision to 8 decimal places

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
Refactor start broadcasting and update precision to 8 decimal places

## Related Issue
N/A

## Motivation and Context
Traffic Incident use case

## How Has This Been Tested?
Local Integration testing

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: carma-messenger-core/traffic_incident/src/traffic_incident_node.cpp [hunk 1]
```diff
@@ -61,37 +74,11 @@
...
-        }catch(...){
-              //in case any exception
-              resp.success = false;
-              return false;
-        }    
+      //update instance variables with incoming request params
+      traffic_worker_.setMinGap(req.minimum_gap);
+      traffic_worker_.setDownTrack(req.down_track);
+      traffic_worker_.setUpTrack(req.up_track);
+      traffic_worker_.setAdvisorySpeed(req.advisory_speed);
```
Comment (msmcconnell) [inline carma-messenger-core/traffic_incident/src/traffic_incident_node.cpp:81]: You should return true at the end of the service callback.

File: carma-messenger-core/traffic_incident/src/traffic_incident_node.cpp [hunk 2]
```diff
@@ -61,37 +74,11 @@
...
-              //construct local mobilityOperation msg
-              cav_msgs::MobilityOperation traffic_mobility_msg = traffic_worker_.mobilityMessageGenerator(traffic_worker_.getPinPoint());
-              
-              if(traffic_worker_.getDownTrack()>0  && traffic_worker_.getUpTrack()>0 && traffic_worker_.getMinGap() > 0) 
-              {
-                //start constantly broadcasting mobilityOperation msg
-                traffic_mobility_operation_pub_.publish(traffic_mobility_msg);
-
-                //return service response true 
-                resp.success=true;
```
Comment (msmcconnell) [inline carma-messenger-core/traffic_incident/src/traffic_incident_node.cpp:82]: You should still include the resp.success = true; line.

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: carma-messenger-core/traffic_incident/src/traffic_incident_node.cpp (hunk 2 above)
Comment (dan-du-car) [reply carma-messenger-core/traffic_incident/src/traffic_incident_node.cpp:82]: https://github.com/usdot-fhwa-stol/carma-messenger/pull/84/commits/d96950eae103d8a6630d50903cd1175a74af6208

File: carma-messenger-core/traffic_incident/src/traffic_incident_node.cpp (hunk 1 above)
Comment (dan-du-car) [reply carma-messenger-core/traffic_incident/src/traffic_incident_node.cpp:81]: https://github.com/usdot-fhwa-stol/carma-messenger/pull/84/commits/d96950eae103d8a6630d50903cd1175a74af6208

**Review by msmcconnell — APPROVED** [review]

**Pull request merged**